pkg/kubectl/cmd/get_test.go

```diff
@@ -162,9 +162,9 @@ func TestGetUnknownSchemaObjectListGeneric(t *testing.T) {
 		},
 		"handles second specific version": {
 			output: "unlikelyversion",
-			list:   "v1beta3",
+			list:   "v1",
 			obj1:   "unlikelyversion", // doesn't have v1beta3
-			obj2:   "v1beta3",         // version of the API response
+			obj2:   "v1",              // version of the API response
```

Comment:
Nevermind the previous comment. I read through more of the test.

Sigh. I wish (again) I could leave comments on unchanged lines.

It sounds like what we should do is:
* change `apiCodec := runtime.CodecFor(api.Scheme, "v1beta3")` to use `testapi.Version()` -- this codec is used to return the ReplicationController
* continue to use `latest.Version` in `tf.ClientConfig = &client.Config{Version: latest.Version}`
* rename type/foo to testtype/foo
* rename output, list, obj1, and obj2 to outputVersion, listVersion, testtypeVersion, and rcVersion
* replace the strings in the testCases table with expressions that make it clear which version is expected and why: testVersion (testapi.Version()), clientVersion (latest.Version), testfactoryVersion ("unlikelyversion")

My understanding of the behavior, which should be documented in a code comment is:
* objects using unrecognized schemes will always be returned using that scheme/version ("unlikelyversion")
* if the specified output-version is a recognized, valid Scheme, then the list should use that scheme, and otherwise it will default to the client version, latest.Version in this case
* if the specified output-version is a recognized, valid Scheme, in which the requested object (replicationcontrollers) can be represented, then the object should be returned using that version, and otherwise should use the codec version returned by the server (testapi.Version in this case)